retry/retry.go

```diff
@@ -11,27 +11,31 @@ import (
 	"github.com/pkg/errors"
 )
 
-const taskSystemName = "task"
+const (
+	taskSystemName = "task"
+	maxWaitTime    = 15 * float64(time.Minute) // 15 minutes
```

Comment:
Please put the unit in the variable name, when `maxWaitTime` is later used, it is not clear if it is in mins or seconds.
Also, is it possible to use a type that is better suited for time such as duration?

Reply:
I guess it's technically nanoseconds. I didn't want to change to use duration since the WaitTime(*) algorithm uses floats

Reply:
We don't have to change the method signature, we can define this a duration, and get the total nanoseconds from it right where we call it (assuming we don't call it in many places)